File: src/main/java/pl/training/workshop/m6/s09_observer/step2/LoyaltyPoints.java
package pl.training.workshop.m6.s09_observer.step2;

import pl.training.workshop.m6.s09_observer.LoyaltyProgram;

/** Krok 2: obserwator - 1 punkt za każde pełne 10.00. */
public record LoyaltyPoints(LoyaltyProgram loyalty) implements PaymentListener {
    @Override
    public void onPaid(ReservationPaid event) {
        loyalty.addPoints(event.email(), event.amount().amount().intValue() / 10);
    }
}

File: csharp/src/Training.Workshop/M6/S09Observer/Step2/LoyaltyPoints.cs
namespace Training.Workshop.M6.S09Observer.Step2;

/// <summary>Krok 2: obserwator - 1 punkt za każde pełne 10.00.</summary>
public sealed record LoyaltyPoints(ILoyaltyProgram Loyalty) : IPaymentListener
{
    public void OnPaid(ReservationPaid paidEvent)
    {
        Loyalty.AddPoints(paidEvent.Email, (int)paidEvent.Amount.Amount / 10);
    }
}
